src/snowflake/cli/_plugins/nativeapp/entities/application_package.py

```diff
@@ -648,8 +642,8 @@ def _deploy(
                 print_diff=print_diff,
             )
 
-            if run_post_deploy_hooks:
-                self.execute_post_deploy_hooks()
+        if run_post_deploy_hooks:
+            self.execute_post_deploy_hooks()
```

Comment:
Since this will be calling use_role through the sql facade, we don't need to nest it inside the above context manager